### PR TITLE
feat: [CDS-80410]: support type "http" for Http step and create HttpStepInfoV1

### PR DESCRIPTION
**Pull request by shalini-agr**
## Describe your changes

**Changes:** Created Http Step info v1 and step parameters v1 to deserialize v1 http yaml. Also, created HttpStepV1, as the step type has been changed to 'http' in v1.

**Tested:** Tested locally that HTTP step with v1 and v0 yaml is running fine and the expressions of this step are also working.

## Checklist
- [ ] I've documented the changes in the PR description.
- [ ] I've tested this change either in PR or local environment.
- [ ] If hashcheck failed I followed [the checklist](https://harness.atlassian.net/wiki/spaces/DEL/pages/21016838831/PR+Codebasehash+Check+merge+checklist) and updated this PR description with my answers to make sure I'm not introducing any breaking changes.

## Comment Triggers
<details>
  <summary>Build triggers</summary>

- Feature build: `trigger feature-build`
- 
  Specific builds
  - `trigger manager`
  - `trigger dms`
  - `trigger ng_manager`
  - `trigger cvng `
  - `trigger cmdlib`
  - `trigger template_svc`
  - `trigger events_fmwrk_monitor`
  - `trigger event_server`
  - `trigger change_data_capture`
  -  Trigger multiple builds together: For eg: `trigger dms, manager`
- Immutable delegate `trigger publish-delegate`
</details>

## [Latest PR Check Triggers](https://github.com/harness/harness-core/blob/develop/.github/pull_request_template.md)

<details>
  <summary>PR Check triggers</summary>
You can run multiple PR check triggers by comma separating them in a single comment. e.g. `trigger ut0, ut1`

- Compile: `trigger compile`
- CodeformatCheckstyle: `trigger checkstylecodeformat`
  - CodeFormat: `trigger codeformat`
  - Checkstyle: `trigger checkstyle`
- MessageMetadata: `trigger messagecheck`
- File-Permission-Check: `trigger checkpermission`
- Recency: `trigger recency`
- BuildNumberMetadata: `trigger buildnum`
- Trigger CommonChecks: `trigger commonchecks`
- PMD: `trigger pmd`
- Copyright Check: `trigger copyrightcheck`
- Feature Name Check: `trigger featurenamecheck`
- UnitTests-ALL: `trigger uts`
- UnitTests-0: `trigger ut0`
- UnitTests-1: `trigger ut1`
- UnitTests-2: `trigger ut2`
- UnitTests-3: `trigger ut3`
- UnitTests-4: `trigger ut4`
- UnitTests-5: `trigger ut5`
- UnitTests-6: `trigger ut6`
- UnitTests-7: `trigger ut7`
- UnitTests-8: `trigger ut8`
- UnitTests-9: `trigger ut9`
- CodeBaseHash: `trigger codebasehash`
- CodeFormatCheckstyle: `trigger checkstylecodeformat`
- SonarScan: `trigger ss`
- GitLeaks: `trigger gitleaks`
- Trigger all Checks: `trigger smartchecks`
- Go Build: `trigger gobuild`
- Validate_Reviews: `trigger review`
- Module Dependency Check: `trigger mdc`
</details>

## PR check failures and solutions
https://harness.atlassian.net/wiki/spaces/BT/pages/21106884744/PR+Checks+-+Failures+and+Solutions

## [Contributor license agreement](https://github.com/harness/harness-core/blob/develop/CONTRIBUTOR_LICENSE_AGREEMENT.md)

- - -
This change is [Reviewable](https://reviewable.io/reviews/harness/harness-core/53496)

### CODE REVIEW

**Review by archit-harness — COMMENTED** [review]
> Reviewed 1 of 8 files at r1, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/53496)* status: 8 file(s) to review. 1 change requests. 0 of 2 approvals obtained.

File: pipeline-service/modules/orchestration-steps/src/main/java/io/harness/plancreator/steps/http/v1/HTTPStepVariableCreatorV1.java [hunk 1]
```diff
@@ -0,0 +1,30 @@
...
+import io.harness.annotations.dev.OwnedBy;
+import io.harness.plancreator.steps.http.HttpStepNode;
+import io.harness.pms.sdk.core.pipeline.variables.GenericStepVariableCreator;
+import io.harness.steps.StepSpecTypeConstantsV1;
+
+import java.util.Collections;
+import java.util.Set;
+
+@OwnedBy(HarnessTeam.PIPELINE)
+public class HTTPStepVariableCreatorV1 extends GenericStepVariableCreator<HttpStepNode> {
```
Comment (utkarshchoubey-harness) [inline pipeline-service/modules/orchestration-steps/src/main/java/io/harness/plancreator/steps/http/v1/HTTPStepVariableCreatorV1.java:20]: shouldn't this be HttpStepNodeV1 ?

**Review by utkarshchoubey-harness — CHANGES_REQUESTED** [review]

**Review by hextechpal — COMMENTED** [review]
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/53496)* status: 8 file(s) to review. 2 change requests. 1. unresolved discussions. 0 of 2 approvals obtained. 

___
*[`pipeline-service/modules/orchestration-steps/src/main/java/io/harness/plancreator/steps/http/v1/HTTPStepVariableCreatorV1.java` line 20 at r3](https://reviewable.io/reviews/harness/harness-core/53496#gh-1348370767:-Ng3bXHx3RDM6xyHykLj:b-8asjoj) ([raw file](https://github.com/harness/harness-core/blob/baa38c65a8d8499aa7c7043a3250de2688c7c10e/pipeline-service/modules/orchestration-steps/src/main/java/io/harness/plancreator/steps/http/v1/HTTPStepVariableCreatorV1.java#L20)):*
<details><summary><i>Previously, utkarshchoubey-harness wrote…</i></summary><blockquote>

shouldn't this be HttpStepNodeV1 ?
</blockquote></details>

@archit-harness @shalini-agr should we change our variable creator too to follow the plan creator route and expect yaml rather than POJOs

**Review by archit-harness — CHANGES_REQUESTED** [review]
> Reviewed 15 of 15 files at r4, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/53496)* status: 17 file(s) to review. 2 change requests. 0 of 2 approvals obtained. 

___
*[`pipeline-service/modules/orchestration-steps/contracts/src/main/java/io/harness/plancreator/steps/http/v1/HttpStepInfoV1.java` line 31 at r1](https://reviewable.io/reviews/harness/harness-core/53496#-NfQbzxY3n443nUxf2Um:-NgJKnyD3RrbqxIEcvJE:bb7f2mf) ([raw file](https://github.com/harness/harness-core/blob/12f3354bd718d753c0d48a6c934c401e75170c30/pipeline-service/modules/orchestration-steps/contracts/src/main/java/io/harness/plancreator/steps/http/v1/HttpStepInfoV1.java#L31)):*
<details><summary><i>Previously, archit-harness (Archit Singla) wrote…</i></summary><blockquote>

why ng variable ?
</blockquote></details>

done
___
*[`pipeline-service/modules/orchestration-steps/src/main/java/io/harness/plancreator/steps/http/v1/HTTPStepVariableCreatorV1.java` line 20 at r3](https://reviewable.io/reviews/harness/harness-core/53496#gh-1348370767:-NgJKvQ-AXykBnrdGRa4:bnaqhjw) ([raw file](https://github.com/harness/harness-core/blob/baa38c65a8d8499aa7c7043a3250de2688c7c10e/pipeline-service/modules/orchestration-steps/src/main/java/io/harness/plancreator/steps/http/v1/HTTPStepVariableCreatorV1.java#L20)):*
<details><summary><i>Previously, hextechpal (Prashant Pal) wrote…</i></summary><blockquote>

@archit-harness @shalini-agr should we change our variable creator too to follow the plan creator route and expect yaml rather than POJOs
</blockquote></details>

@hextechpal yes we will do that, @shalini-agr can you change this, but just FYI -> we are planning once we have resource to remove variables creators all together
___
*[`pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/http/HttpStepUtils.java` line 46 at r4](https://reviewable.io/reviews/harness/harness-core/53496#-NgJLDCgBO7c1k2VcD0M:-NgJLDCgBO7c1k2VcD0N:b-2cc6vt) ([raw file](https://github.com/harness/harness-core/blob/69ffc99f436535a811b88dbc168fdbca0ec17172/pipeline-service/modules/orchestration-steps/src/main/java/io/harness/steps/http/HttpStepUtils.java#L46)):*
> ```Java
> 
> @Slf4j
> public class HttpStepUtils {
> ```

add codepulse annotation

**Review by archit-harness — APPROVED** [review]
> Reviewed 2 of 2 files at r5, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/53496)* status: 17 file(s) to review. 1 change requests. 1 of 2 approvals obtained.

**Comment by shalini-agr** [comment]
trigger uts

**Comment by shalini-agr** [comment]
trigger uts

**Review by archit-harness — COMMENTED** [review]
> Reviewed 1 of 3 files at r6, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/53496)* status: 18 file(s) to review. 1 change requests. 0 of 2 approvals obtained.

**Review by utkarshchoubey-harness — APPROVED** [review]

**Comment by shalini-agr** [comment]
trigger uts

**Review by archit-harness — APPROVED** [review]
> Reviewed 8 of 8 files at r7, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/53496)* status: 22 file(s) to review. 1 of 2 approvals obtained.

**Comment by shalini-agr** [comment]
trigger uts

**Comment by shalini-agr** [comment]
trigger ut8

**Review by archit-harness — APPROVED** [review]
> Reviewed 4 of 4 files at r8, all commit messages.
> *[Reviewable](https://reviewable.io/reviews/harness/harness-core/53496)* status: 26 file(s) to review.

**Comment by shalini-agr** [comment]
trigger review

**Pull request merged**